### PR TITLE
refactor(scrollbar): spread new scroll component

### PR DESCRIPTION
**Pull request by ThomBos**
**What this PR does** 📖
Substitutes old scroll component with a new native one
**Which issue(s) this PR fixes** 🔨
AP-1604

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

File: assets/styles/base.less [hunk 1]
```diff
@@ -613,6 +613,11 @@ video {
 .editable-container {
   .editable-input {
     .chat-row-content {
+      display: flex;
+      flex-direction: column;
+      justify-content: flex-start;
+      align-items: flex-start;
+      box-sizing: border-box;
```
Comment (molimauro) [inline assets/styles/base.less:620]: why this?

File: components/views/chat/chatbar/Chatbar.less [hunk 2]
```diff
@@ -96,11 +96,36 @@
...
+      white-space: normal;
+      display: flex;
+      flex-direction: row;
+      justify-content: flex-start;
+      align-items: center;
+      padding-top: @light-spacing;
+      padding-bottom: @light-spacing;
       min-height: 48px;
-      max-height: 50vh;
+      max-height: 25vh;
```
Comment (molimauro) [inline components/views/chat/chatbar/Chatbar.less:128]: revert

File: components/views/chat/chatbar/Chatbar.less [hunk 3]
```diff
@@ -96,11 +96,36 @@
...
+        padding-left: @light-spacing;
+      }
+      p {
+        font-size: @text-size !important;
+        .chatbar-tag {
+          &:extend(.round-corners);
+          background: @midground;
+          padding: @xlight-spacing;
+        }
+      }
```
Comment (molimauro) [inline components/views/chat/chatbar/Chatbar.less:116]: revert

File: components/views/chat/chatbar/Chatbar.vue [hunk 4]
```diff
@@ -357,24 +357,3 @@ export default Vue.extend({
...
-  p {
-    font-size: @text-size !important;
-    .chatbar-tag {
-      &:extend(.round-corners);
-      background: @midground;
-      padding: @xlight-spacing;
-    }
-  }
-}
-</style>
```
Comment (molimauro) [inline components/views/chat/chatbar/Chatbar.vue:380]: revert

**Comment by molimauro** [comment]
The editable css is so easy to break that its better to ask me before touching it. Revert please the updates related to that thanks!

File: assets/styles/base.less (hunk 1 above)
Comment (ThomBos) [reply assets/styles/base.less:620]: can't tell you... i don't remember if this style was usefull when i tried removing the wrapper from the new scroll component or is needed to fix its position. i can't test it rn due to textile issues

File: components/views/chat/chatbar/Chatbar.less [hunk 5]
```diff
@@ -96,11 +96,36 @@
...
     }
 
     .editable-container {
+      white-space: normal;
+      display: flex;
+      flex-direction: row;
+      justify-content: flex-start;
+      align-items: center;
+      padding-top: @light-spacing;
+      padding-bottom: @light-spacing;
```
[the remark this answers is not in the document]
Comment (ThomBos) [reply components/views/chat/chatbar/Chatbar.less:126]: reverting this i've to go back to the previous scrollbar component. the problem is that the chatbar component has 2 things overlapped and needs this class to be sent to the new scrollbar component to wrap chatbar properly

[the remark this answers is not in the document]
Comment (molimauro) [reply components/views/chat/chatbar/Chatbar.less:126]: But why white-space: normal;?

[the remark this answers is not in the document]
Comment (ThomBos) [reply components/views/chat/chatbar/Chatbar.less:126]: > But why `white-space: normal;`?
it's a mistake

[the remark this answers is not in the document]
Comment (molimauro) [reply components/views/chat/chatbar/Chatbar.less:126]: Remove it and revert all the others then ill test thanks

File: components/views/chat/chatbar/Chatbar.less (hunk 2 above)
Comment (ThomBos) [reply components/views/chat/chatbar/Chatbar.less:128]: this 50vh has some overlapping problems (test it with the search feature on the toolbar)

File: components/views/chat/chatbar/Chatbar.less [hunk 6]
```diff
@@ -99,6 +99,11 @@
     }
 
     .editable-container {
+      white-space: normal;
```
Comment (molimauro) [inline components/views/chat/chatbar/Chatbar.less:102]: still seeing this

File: assets/styles/base.less (hunk 1 above)
Comment (ThomBos) [reply assets/styles/base.less:620]: yeah i'm testing it

**Comment by molimauro** [comment]
Seems much better!! :) I'll review the remaining code 💥

**Review by molimauro — APPROVED** [review]

**Comment by molimauro** [comment]
Great! How much components are left with the old scollbar library?

**Comment by ThomBos** [comment]
> Great! How much components are left with the old scollbar library?

like 2 or 3

**Comment by molimauro** [comment]
Ok perfect, we need to talk about them and understand how to purge them

**Review by stavares843 — CHANGES_REQUESTED** [review]
dev

Captura de ecrã 2022-05-30, às 16 10 45


this branch

Captura de ecrã 2022-05-30, às 16 09 40



is missing the pagination numbers on the bottom when searching

**Comment by ThomBos** [comment]
> is missing the pagination numbers on the bottom when searching

i'm fixing it rn, but i have some questions about that...
should we have scroll and pagination here?? isn't it confusing having both?

**Review by molimauro — APPROVED** [review]

**Review by molimauro — APPROVED** [review]

**Comment by ThomBos** [comment]
are submodules still there? (because i don't see them😅) @molimauro

**Comment by molimauro** [comment]
Doesnt seem so

**Comment by phillsatellite** [comment]
Tested: Looked good! I could only find one little issue when user is trying to send a bulk amount of files the scroll is gone and the files will stack up on each other 
Simulator Screen Shot - iPhone 13 Pro Max - 2022-06-03 at 09 54 44

https://user-images.githubusercontent.com/93608357/171869467-b1e9d52d-a798-4a04-9238-ad755e7aa0d6.mov

**Comment by ThomBos** [comment]
> Tested: Looked good! I could only find one little issue when user is trying to send a bulk amount of files the scroll is gone and the files will stack up on each other 
Thanks Phill! 👍 
The correct behaviour should be showing just the last image, while to see the others the user has to scroll down, right?

**Comment by ThomBos** [comment]
should work now! 👍

**Comment by phillsatellite** [comment]
Tested: Looks like one more tiny bug appeared from recent changes that I couldn't replicate on dev or test.satellite. If the user adjust the browser size it will eventually start to cover up the names on the preview 

Video shows this branch vs test.satellite (I tested dev branch but just used test.satellite for the screen recording)


https://user-images.githubusercontent.com/93608357/171918940-87a8a0ef-a268-4567-8c10-46151218776f.mov

**Review by iltumio — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Comment by phillsatellite** [comment]
Note - Was able to test on desktop and android but wasnt able to get past linking satellites on iOS

**Comment by ThomBos** [comment]
Need this PR to be merged before opening another PR for AP-1730

**Comment by stavares843** [comment]
was only fixing the conflict, and bot removed the label 😂 will retest and merge

File: components/ui/SimpleScroll/SimpleScroll.vue [hunk 7]
```diff
@@ -25,6 +25,21 @@ export default Vue.extend({
       default: '',
       required: false,
     },
+    scrollTopOnChange: {
+      type: [String, Boolean, Number, Array],
```
Comment (molimauro) [inline components/ui/SimpleScroll/SimpleScroll.vue:29]: whats this type?

**Review by stavares843 — APPROVED** [review]

**Pull request merged**